torch/csrc/jit/script/compiler.cpp

```diff
@@ -1396,17 +1396,64 @@ struct to_ir {
       const Ident& target,
       const List<Expr>& args,
       const List<Stmt>& body) {
-    // TODO: start, stop, step loop
-    if (args.size() != 1) {
+    Value *end_val = nullptr, *start_val = nullptr, *step_val = nullptr;
+    bool isSimpleRange = (args.size() == 1);
+    if (isSimpleRange) {
+      end_val = ensureInt(range, emitExpr(args[0]));
+      start_val = end_val->owningGraph()->insertConstant(0);
+      step_val = end_val->owningGraph()->insertConstant(1);
+    } else if (args.size() == 2) {
+      start_val = ensureInt(range, emitExpr(args[0]));
+      end_val = ensureInt(range, emitExpr(args[1]));
+      step_val = end_val->owningGraph()->insertConstant(1);
+    } else if (args.size() == 3) {
+      start_val = ensureInt(range, emitExpr(args[0]));
+      end_val = ensureInt(range, emitExpr(args[1]));
+      step_val = ensureInt(range, emitExpr(args[2]));
+    } else {
       throw ErrorReport(range)
-          << "range() expects 1 argument but got " << args.size();
+          << "range() expected at most 3 arguments, got " << args.size();
     }
-    auto max_trip_count_val = ensureInt(range, emitExpr(args[0]));
     const auto& ident_name = target.name();
-    auto assigner = [ident_name, range](
+    assert(end_val != nullptr);
```

Comment:
We have our own assert macros that print out a stack trace along with the error, `AT_ERROR(cond)` and `AT_CHECK(cond, message...)`